Src/GoogleApis.Auth/OAuth2/Flows/GoogleAuthorizationCodeFlow.cs

```diff
@@ -72,12 +78,25 @@ public override AuthorizationCodeRequestUrl CreateAuthorizationCodeRequest(strin
             await DeleteTokenAsync(userId, taskCancellationToken);
         }
 
+        public override bool ShouldForceTokenRetrieval() 
+        {
+            if (includeGrantedScopes.HasValue && includeGrantedScopes.Value == true)

```

Comment:
Done, pending the bool vs bool? thoughts above. You kids and your crazy shorthanded logic =) Good catch.